File: ios/Kwift Template/swiftResources/layouts/ViewPagerDemoXml.swift
//
// ViewPagerDemoXml.swift
// Created by Kwift XML
//

import UIKit
import Kwift

class ViewPagerDemoXml {
    
    unowned var xmlRoot: UIView!
    func setup(_ dependency: ViewDependency) -> UIView {
        let view = FrameLayout(frame: .zero)
        view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
        view.addSubview(
            UICollectionView(frame: .zero, collectionViewLayout: ViewPagerLayout()),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 0 + 0, left: 0 + 0, bottom: 0 + 0, right: 0 + 0),
            gravity: .fillFill
        ) { view in 
            self.viewPager = view
            view.canCancelContentTouches = false
            view.showsHorizontalScrollIndicator = false
            view.backgroundColor = .clear
        }
        
        view.addSubview(
            UIPageControl(frame: .zero),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 8 + 0, left: 8 + 0, bottom: 8 + 0, right: 8 + 0),
            gravity: .bottomCenter
        ) { view in 
            self.viewPagerIndicator = view
            view.currentPageIndicatorTintColor = ResourcesColors.colorPrimary
            view.pageIndicatorTintColor = UIColor(argb: 0xFF888888)
        }
        
        xmlRoot = view
        return view
    }
    
    unowned var viewPagerIndicator: UIPageControl!
    unowned var viewPager: UICollectionView!
    
}

File: android/app/src/main/java/com/lightningkite/kwifttemplate/layouts/ViewPagerDemoXml.kt
//
// ViewPagerDemoXml.swift
// Created by Kwift XML Android
//
package com.lightningkite.kwifttemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.kwift.views.*
import com.lightningkite.kwifttemplate.R

class ViewPagerDemoXml {

    lateinit var viewPager: androidx.viewpager.widget.ViewPager
    lateinit var viewPagerIndicator: com.rd.PageIndicatorView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.view_pager_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        viewPager = view.findViewById<androidx.viewpager.widget.ViewPager>(R.id.viewPager)
        viewPagerIndicator = view.findViewById<com.rd.PageIndicatorView>(R.id.viewPagerIndicator)
        
        
        return view
    }
}